setup.py

```diff
@@ -3,30 +3,40 @@
 
 import os
 
-import numpy
-from Cython.Distutils import build_ext
 from setuptools import Extension, find_packages, setup
+from setuptools.command.build_ext import build_ext as _build_ext
+
+# Test if compiling with cython or using the C source
+try:
+    from Cython.Distutils import build_ext as _build_ext
+except ImportError:
+    USE_CYTHON = False
+else:
+    USE_CYTHON = True
+
+print('Using Cython {}'.format(USE_CYTHON))
+ext = '.pyx' if USE_CYTHON else '.c'
 
 
 def c_name_from_path(location, name):
     return os.path.join(location, name).replace('/', '.')
 
 
-with open('requirements.txt') as requirements_file:
-    requirements = requirements_file.read()
+class build_ext(_build_ext):
+    def finalize_options(self):
+        _build_ext.finalize_options(self)
+        # Prevent numpy from thinking it is still in its setup process:
+        __builtins__.__NUMPY_SETUP__ = False
+        import numpy
+        self.include_dirs.append(numpy.get_include())
```

Comment:
This is hieroglyphics to me. I guess what I really would like to know is what is this improving, specifically?

Reply:
Haha, me too. I just copied it from SO and it works. It's making setup.py load numpy if we're compiling the .pyx (cython) to C. This gets around an import numpy at the top, which will break smrf installing if numpy isn't already installed.

Reply:
Ok. I get it. So at what point does numpy get installed if it isn't already?

Reply:
Where is the import? From within Cython?

Reply:
numpy will be installed as a dependency to SMRF, it's just not needed now to install SMRF, which was the hangup.

All numpy was needed for in setup.py was `numpy.get_include()` for it's headers. Other than that, it isn't needed. If we're compiling the .pyx to .c, then numpy will be used in the `build_ext`.